File: mifospay/src/main/java/org/mifos/mobilewallet/mifospay/password/EditPasswordContract.java
package org.mifos.mobilewallet.mifospay.password;

import android.content.Context;

import org.mifos.mobilewallet.mifospay.base.BasePresenter;
import org.mifos.mobilewallet.mifospay.base.BaseView;

/**
 * Created by ankur on 27/June/2018
 */

public interface EditPasswordContract {

    interface EditPasswordPresenter extends BasePresenter {

        void updatePassword(String currentPassword, String newPassword, String newPasswordRepeat);

        void handleSavePasswordButtonStatus(String currentPassword,
                                            String newPassword,
                                            String newPasswordRepeat);
    }

    interface EditPasswordView extends BaseView<EditPasswordPresenter> {

        Context getContext();

        void closeActivity();

        void startProgressBar();

        void stopProgressBar();

        void showError(String msg);

        void enableSavePasswordButton();

        void disableSavePasswordButton();
    }
}

File: mifospay/src/main/java/org/mifos/mobilewallet/mifospay/password/EditPasswordContract.kt
package org.mifos.mobilewallet.mifospay.password

import org.mifos.mobilewallet.mifospay.base.BasePresenter
import org.mifos.mobilewallet.mifospay.base.BaseView

/**
 * Created by ankur on 27/June/2018
 */
interface EditPasswordContract {
    interface EditPasswordPresenter : BasePresenter {
        fun updatePassword(
            currentPassword: String?,
            newPassword: String?,
            newPasswordRepeat: String?
        )

        fun handleSavePasswordButtonStatus(
            currentPassword: String?,
            newPassword: String?,
            newPasswordRepeat: String?
        )
    }

    interface EditPasswordView : BaseView<EditPasswordPresenter?> {
        fun closeActivity()
        fun startProgressBar()
        fun stopProgressBar()
        fun showError(msg: String?)
        fun enableSavePasswordButton()
        fun disableSavePasswordButton()
    }
}